datastore/nipost/nipost.go

```diff
@@ -0,0 +1,125 @@
+package nipost
+
+import (
+	"fmt"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/datastore"
+	"github.com/spacemeshos/go-spacemesh/sql"
+)
+
+func AddChallenge(db *datastore.LocalDB, nodeID types.NodeID, ch *types.NIPostChallenge) error {
+	enc := func(stmt *sql.Statement) {
+		stmt.BindBytes(1, nodeID.Bytes())
+		stmt.BindInt64(2, int64(ch.PublishEpoch))
+		stmt.BindInt64(3, int64(ch.Sequence))
+		stmt.BindBytes(4, ch.PrevATXID.Bytes())
+		stmt.BindBytes(5, ch.PositioningATX.Bytes())
+		if ch.CommitmentATX != nil {
+			stmt.BindBytes(6, ch.CommitmentATX.Bytes())
+		} else {
+			stmt.BindNull(6)
+		}
+		if ch.InitialPost != nil {
+			stmt.BindInt64(7, int64(ch.InitialPost.Nonce))
+			stmt.BindBytes(8, ch.InitialPost.Indices)
+			stmt.BindInt64(9, int64(ch.InitialPost.Pow))
+		} else {
+			stmt.BindInt64(7, -1)
+			stmt.BindNull(8)
+			stmt.BindInt64(9, -1)
```

Comment:
I can bind `NULL` to a nullable value, but `stmt.ColumnInt64` will then return `0` for that value if used and I thought `-1` was a better "not available" value.

Since the other two values are assumed to be `NULL` anyway when `Indices` is `NULL`, it doesn't hurt to use `stmt.BindNull` 🙂